autorest/codegen/serializers/general_serializer.py

```diff
@@ -20,12 +20,27 @@ def serialize_pkgutil_init_file(self) -> str:
 
     def serialize_init_file(self) -> str:
         template = self.env.get_template("init.py.jinja2")
-        return template.render(code_model=self.code_model, async_mode=self.async_mode,)
+        return template.render(code_model=self.code_model, async_mode=self.async_mode)
+
+    def _correct_credential_parameter(self):
+        credential_param = [
+            gp for gp in self.code_model.global_parameters.parameters if isinstance(gp.schema, CredentialSchema)
+        ][0]
```

Comment:
If add-credentials is False, there is no credential parameter, and your [0] will raise